### ISSUE TITLE
Request for a final case result variable 

### ISSUE THREAD

**Issue by Iheanacho2027** [SORMAS-Foundation/SORMAS-Project#1860]
### Situation Description
Currently the SORMAS dashboard considers the latest test result as the final result of a case which isnt a true reflection of the case statistically as the latest result might be a retest for a confirmed case, statistically the number of confirmed cases should align with the total number of positive results on the dashboard as the dashboard should consider only one result per case
### Feature Description
There should be a variable for selecting the final result of a case. I suggest that this be prioritized for the next release 1.39.0 as dashboards across regions in Nigeria are not giving a true reflection of happenings in this current pandemic @MateStrysewskeSym @bernardsilenou

**Comment by bernardsilenou** [comment]
@Iheanacho2027  This is right, we are working on an algorithm to define this final lab result of the case.

**Comment by Iheanacho2027** [comment]
Pls note the variable should be first greyed out as long as a result hasnt been entered for a sample in the case, then should be made editable after one or more results for a sample(s) have been entered

**Comment by bernardsilenou** [comment]
@Iheanacho2027 please if you have any idea, drop here:

**Comment by MateStrysewskeSym** [comment]
From Bernard:

When there are more than one sample for a case and each sample has a final lab result, What should SORMAS do to determine the final lat result of the case?
2,1 Automatically choose the sample with positive result? If there are more than one samples with
positive final sample result, then:
2,2 Ask the user to choose one of the final sample result as the sample result of the case?
2.3 Automatically calculate the majority result (in case there are more than 2 samples) and use this
as the result of the case?

**Comment by MartinWahnschaffeSymeda** [comment]
Highly related to #1843

**Comment by bernardsilenou** [comment]
For the final result of the case, Must we have a final sample result registered in the system?  If yes, then for users to set a case to confirmed we must force them to record a lab results?
If no, then we fall in the same problem of number of rest  results != number of confirmed cases.

But  for final lab result of a sample, I think it should be based on the the recorded lab results  of the sample. We do not allow users to edit the final lab result of a sample without registering a sample result.
Since we currently use one (first sample result) of the samples of a case for automatic classification, maybe we need more refinement before implementing this issue. What do you think?

**Comment by Iheanacho2027** [comment]
Here is my proposal on the algorithm which should be used for the system's selection of the final case result. Since the final case result should be what will determine the classification of the case, then
1) If the result of the initial or first test is negative, the system should select negative as the final case result
2) If the result of any of the tests (if there are multiple tests) is positive, the system should select positive as the final case result

However, the system should allow the user to edit the final case result after selection

**Comment by bernardsilenou** [comment]
Similarly to the final lab result of a sample which is defined as one of all the individual test results of the sample and is chosen by the user each time a new sample result is recorded that differs from the previous one, the final lab result for the case is also one of the final lab results of all the samples of the case.
- The challenge here is that a case can recover, new sample taken and its final lab result is negative. This will the appear as it the case in "NOT_CASE". After a case is initially classified as confirmed case, we need to prevent the case to latter be classified as not a case based on a future negative test result. A confirmed case always stay confirmed.
Here is a general algorithm to calculate the final lab results for entities like cases, contacts and event participants:

1. When a case, contact or event participant is created, Initialize the final entity result to NULL
2. When the first final lab result of the sample(s) associated to the entity is not Null, automatically set this result as the final lab result of the entity.
3. Future final lab results of samples associated to the entity should not trigger step 2 any more after its initial assignment.
4. Allow users to manually change the final result of the entity if they want. This is not the case for final result of samples and it is fine as it is.
5.Instead of using the final lab result of the sample to classify the entity as we currently do, we will then use the final lab result of the entity.
6. In summary, we only use the first final lab result of the samples of the entity to trigger "if user want sormas to automatically classify the entity as confirmed". For subsequent final lab results of other samples associated to the entity, we do not trigger nor automatically update the final lab result of the entity.

**Comment by bernardsilenou** [comment]
@MateStrysewskeSym @markusmann-vg  If you have further refinement, you can add please, this is all from my side. We decided to add this issue on the current print board but i do  not know what was finally decided.
